### PR TITLE
DEV: Split toggle topic close job

### PR DESCRIPTION
**Pull request by martin-brennan**
Splits the `ToggleTopicClosed` job into two distinct `OpenTopic` and `CloseTopic` jobs to make the code clearer. The old job cannot be deleted yet because of outstanding sidekiq schedules, so a todo has been added to do so later this year.

Also replaced mentions of `topic_status_update` with `topic_timer` in some files, because the `topic_status_update` model is obsolete and replaced by topic timer.

Added some shortcut methods for checking if a topic is open/whether a user can change an open topic.

### CODE REVIEW

**Review by CvX — COMMENTED** [review]
Looks like some existing specs need to be updated? 🙂

**Comment by martin-brennan** [comment]
@CvX :man_facepalming: sorry for that, will fix all these shenanigans, should've made sure the tests passed first.

**Comment by martin-brennan** [comment]
@CvX this is all good now.

File: app/jobs/regular/open_topic.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      if !Guardian.new(user).can_open_topic?(topic)
+        topic_timer.destroy!
+        topic.reload
+
+        topic.inherit_auto_close_from_category(timer_type: :close)
+
+        return
+      end
+
+      if topic.open?
```
Comment (lis2) [inline app/jobs/regular/open_topic.rb:26]: can we move that code to line :12 to make it more concise?
Similar to combined logic from close_topic.rb
```
if topic.blank? || topic.open?
```

File: spec/jobs/close_topic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+describe Jobs::CloseTopic do
+  fab!(:admin) { Fabricate(:admin) }
+
+  fab!(:topic) do
+    Fabricate(:topic_timer, user: admin).topic
+  end
+
+  it 'should be able to close a topic' do
+    topic
```
Comment (lis2) [inline spec/jobs/close_topic_spec.rb:13]: I don't think that this call is needed `fab!` will create object anyway

**Review by lis2 — APPROVED** [review]

**Comment by lis2** [comment]
@martin-brennan beside 2 small comments, it looks good to me

**Pull request merged**